pipeline/src/main/java/BC_DATABASE_RDD_ENTRY.java

```diff
@@ -203,5 +208,108 @@ public Integer download( List< String > report )
         }
         return errors;
     }
-}
 
+
+
+/**
+ * Check a database file is present and download if it is not. The method is
+ * synchronized for threads and processes. Check and download is atomic.
+ *
+ * @param       report, list of string reporting the download
+ * @return      number of errors
+ * @see              BC_DATABASE_SETTING
+ * @see              BC_GCP_TOOLS
+*/
+    public synchronized Integer downloadIfAbsent( List< String> report)
+    {
+        Integer errors = 0;
+        String wn = workername();
+
+        for ( BC_NAME_SIZE obj : chunk.files )
+        {
+            String extension = obj.name;
+            String src = build_source_path( extension );
+            String dst = build_worker_path( extension );
+            File f = new File( dst );
+
+            String parent = f.getParent();
+            if (parent != null) {
+                File p = new File(parent);
+                p.mkdirs();
+            }
+
+            File ff = new File( dst + ".lock");
```

Comment:
Does .lock file get deleted somehow?

Reply:
No, it does not need to, because existence of the file does not lock. The file is only used and an external object. We could use the downloaded file for locking, but it is created with the FileOutputStream, so I was afraid that an existing database file may be deleted when we try to check whether it is there.